src/pidigits.rs

```diff
@@ -77,18 +62,21 @@ impl Context {
     }
 }
 impl Iterator for Context {
-    type Item = c_ulong;
-    fn next(&mut self) -> Option<c_ulong> {
+    type Item = u8;
+    fn next(&mut self) -> Option<u8> {
         loop {
             self.next_term();
             if self.num > self.acc { continue; }
             let d = self.extract_digit(3);
             if d != self.extract_digit(4) { continue; }
 
             self.eliminate_digit(d);
-            return Some(d);
+            return Some(d as u8);
         }
     }
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        (std::usize::MAX, None)
+    }
```

Comment:
I don't have an optnion on this change. In one hand, that's a more precise implementation. On the other hand, that's more line, unused if collect is removed, and expose useless complexity.